src/Interpreters/TreeRewriter.cpp

```diff
@@ -604,15 +604,13 @@ std::optional<bool> tryEvaluateConstCondition(ASTPtr expr, ContextPtr context)
 
     Field eval_res;
     DataTypePtr eval_res_type;
-    try
     {
-        std::tie(eval_res, eval_res_type) = evaluateConstantExpression(expr, context);
-    }
-    catch (DB::Exception &)
-    {
-        /// not a constant expression
```

Comment:
Thank you! This type of code is always wrong because there can be a memory limit exception, fault injection, or anything else.

Interesting how we can prevent having this code in other places.